Fix master failover and document same.

cro opened the pull request.

Commit:
Improve Upstart enable/disable handling

* Properly search for manual stanza instead of just override
* Handle "manual" stanzas in both override and regular conf files
* Do not try to disable/enable already enabled services
* Do not overwrite override files (use append) on disable
* Only delete empty override files on enable

Commit:
salt.modules.upstart: fix lint errors

Commit:
Add missing import

anlutro committed:
Fix state.apply for salt-ssh

Closes #24746

Commit:
Add notes to map and gogrid docs -- don't use -P with map files

cro committed:
Fix master failover and add documentation for same.

cro committed:
Merge conflict

jfindlay commented:
Go Go Jenkins!

kev009 committed:
Detect bhyve virtual type for freebsd guests

cachedout commented on salt/minion.py:
```diff
@@ -828,12 +830,17 @@ def eval_master(self,
         # if we have a list of masters, loop through them and be
         # happy with the first one that allows us to connect
-        if isinstance(opts['master'], list):
+        if not opts['master_active_list']:
```
We need to handle the case where this is not set yet. We could probably just make this a `get()` and be OK here.

jacksontj committed:
Don't incur a "_load_all" of the lazy_loader while looking for mod_init.

Since we were checking if mod_init is in the loader dict-- and we still allow for module merging (#20540) we will do a load_all for every module without mod_init. This patch makes it so we load w/e module the function is in, then check if we have loaded a mod_init. This cuts out ~600ms on my local simple state.sls runs

cro committed:
Better check for possible missing master_active_list key

cro committed:
Merge conflict

cro committed:
Missed a doc file

cro commented:
Don't merge this yet, I found a bug.

cro committed:
Bad indent

jacksontj committed:
pylint

pruiz committed:
Fix regular expression so it now matches packages with '.' or '-' at pkg name.

Commit:
adding states/ssh_auth unit test case.

cachedout commented:
@cro Still holding on this?

cro commented:
Yeah, had more communication with the customer over the weekend & he brought up a perfectly reasonable use case that stack traces with this PR.

dkiser committed:
Fix for salt-cloud when ssh key used to auth and using sudo.

Fixes #24870

dkiser committed:
Fix pylint for #24874.

dkiser committed:
Fix key error for the PR to fix #24870.

dkiser committed:
Fix to allow password for salt-cloud to be set outside of a vm specific context.

Fixes #24871.

cro committed:
Factor in syndics.  Syndics will not failover (yet).

cro commented:
Closing in favor of #24902.
